src/api/contracts.ts

```diff
@@ -125,6 +125,43 @@ if (process.env.FE_CONDITIONAL_ENV === 'development') {
   }
 }
 
+// in CLAIM_ONLY mode use different contract addresses
+if (process.env.FE_CONDITIONAL_ENV === 'production' && process.env.CLAIM_ONLY) {
+  const localForage = require('localforage')
+
+  const grabOldDXNetworksAndSet = async () => {
+    const ALL_OLD_CONTRACT_ADDRESSES = require('../../test/networks-old')
+    const [CONTRACT_ADDRESSES_TO_USE] = await Promise.all([
+      localForage.getItem('CONTRACT_ADDRESSES_TO_USE'),
+      localForage.setItem('ALL_OLD_CONTRACT_ADDRESSES', ALL_OLD_CONTRACT_ADDRESSES),
+    ])
+
+    if (!CONTRACT_ADDRESSES_TO_USE) {
+      // from networks-old - old versions of DX to grab addresses
+      const latestVersion = Object.keys(ALL_OLD_CONTRACT_ADDRESSES)[0]
+      await Promise.all([
+        localForage.setItem('ALL_OLD_CONTRACT_ADDRESSES', ALL_OLD_CONTRACT_ADDRESSES),
+        localForage.setItem('CONTRACT_ADDRESSES_TO_USE', ALL_OLD_CONTRACT_ADDRESSES[latestVersion]),
+      ])
+      return ALL_OLD_CONTRACT_ADDRESSES[latestVersion]
+    }
+    console.debug(CONTRACT_ADDRESSES_TO_USE)
+    return CONTRACT_ADDRESSES_TO_USE
+  }
+
+  grabOldDXNetworksAndSet()
+  .then((networks: any) => {
```

Comment:
This whole block and especially this function needs to be run before `contr.deployed()`, better yet before `.setProvider()`. Otherwise we'll have a race condition and who knows which addresses will get applied.

First line in `function init()` is a good place